src/layoutparser/ocr/tesseract_agent.py

```diff
@@ -89,10 +90,13 @@ def _detect(self, img_content):
         res["text"] = pytesseract.image_to_string(
             img_content, lang=self.lang, **self.configs
         )
-        _data = pytesseract.image_to_data(img_content, lang=self.lang, **self.configs)
-        res["data"] = pd.read_csv(
-            io.StringIO(_data), quoting=csv.QUOTE_NONE, encoding="utf-8", sep="\t"
-        )
+        _data = pytesseract.image_to_data(img_content, lang=self.lang, output_type=Output.DICT, **self.configs)
+        _df = pd.DataFrame(_data, dtype=str)
+        _cols = list(_df.columns())
+        _cols.remove('text')
+        for col in _cols:
+            _df[col] = _df[col].astype(int)
+        res['data'] = _df
```

Comment:
Can you try the following code: 

```
_data = pytesseract.image_to_data(img_content, lang=self.lang, **self.configs)
df = pd.read_csv(
   io.StringIO(_data), quoting=csv.QUOTE_NONE, encoding="utf-8", sep="\t"
)
df['text'] = df['text'].astype('str')
res["data"] = df
```

Reply:
@lolipopshock  sorry it does not, I have tried this
and ya i get it, the for loop and all that stuff looks ugly :)

here's the screenshot 
![layout_parse](https://user-images.githubusercontent.com/59497032/140306290-36054c21-1e30-4dd0-9be8-0047ae619d16.png)

Reply:
I see -- it's the issue from floating point numbers `.0` right?

Reply:
Yes